### PR TITLE
feat: persisted expand collapse for explorer panels

### PR DESCRIPTION
**Pull request by itssharmasandeep**
## Description
It will provide preference persistence for panels/sections, present in the explorer on expand and collapse.

### Testing
Local testing done.

### Checklist:
- [x] My changes generate no new warnings

### CODE REVIEW

File: projects/observability/src/pages/explorer/explorer.component.ts [hunk 1]
```diff
@@ -146,7 +154,16 @@ export class ExplorerComponent {
     this.vizDashboard$ = this.explorerDashboardBuilder.visualizationDashboard$;
     this.initialState$ = activatedRoute.queryParamMap.pipe(
       take(1),
-      map(paramMap => this.mapToInitialState(paramMap))
+      map(paramMap => this.mapToInitialState(paramMap)),
+      tap(initialState => {
```
Comment (aaron-steinfeld) [inline projects/observability/src/pages/explorer/explorer.component.ts:158]: we can do this without tap, please just use distinct observables.

File: projects/observability/src/pages/explorer/explorer.component.ts [hunk 2]
```diff
@@ -146,7 +154,16 @@ export class ExplorerComponent {
...
+        // Updating initial visualization and results expanded state
+        this.visualizationExpanded = initialState.visualizationExpanded ?? true;
+        this.resultsExpanded = initialState.resultsExpanded ?? true;
+        this.navigationService.addQueryParametersToUrl({
```
Comment (aaron-steinfeld) [inline projects/observability/src/pages/explorer/explorer.component.ts:162]: I think there was a misunderstanding of the ask here - the expand/collapse shouldn't be in the URL, it should be persisted as a preference (see usages of preference service), similar to how table sorts and columns work.

Comment (itssharmasandeep) [reply projects/observability/src/pages/explorer/explorer.component.ts:162]: Can someone edit these preferences?

Comment (aaron-steinfeld) [reply projects/observability/src/pages/explorer/explorer.component.ts:162]: Currently, they're only indirectly exposed. A user's actions update their preferences. Eventually, some may be put in front of a UI (not things like expand collapse but formats, colors etc.)

The simplest example of interaction with preferences is probably the expand/collapse behavior of the left nav in `hypertrace-ui/src/app/shared/navigation/navigation.component.ts`

File: projects/observability/src/pages/explorer/explorer.component.ts (hunk 1 above)
Comment (itssharmasandeep) [reply projects/observability/src/pages/explorer/explorer.component.ts]: Done!!

File: projects/observability/src/pages/explorer/explorer.component.ts (hunk 2 above)
Comment (itssharmasandeep) [reply projects/observability/src/pages/explorer/explorer.component.ts]: Done!!

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**